src/main/java/com/aws/greengrass/certificatemanager/certificate/CISShadowMonitor.java

```diff
@@ -205,59 +208,73 @@ private void subscribeToShadowTopics() throws InterruptedException {
     @SuppressWarnings("PMD.AvoidCatchingGenericException")
     private synchronized void handleNewCloudVersion(int newVersion) {
         if (newVersion == lastVersion) {
-            LOGGER.atInfo().kv("version", newVersion).log("Already processed version. Skipping cert re-generation");
-            reportVersion(newVersion);
+            LOGGER.atInfo().kv(VERSION, newVersion).log("Already processed version. Skipping cert re-generation");
```

Comment:
The fact that this is getting called means there is a shadow delta so you'll want to attempt to re-report